clevertap-core/src/main/java/com/clevertap/android/sdk/cryption/CryptMigrator.kt

```diff
@@ -0,0 +1,292 @@
+                    newGuidJsonObj.toString()
+                )
+                config.logger.verbose(
+                    config.accountId,
+                    "Cached GUIDs migrated successfully: [${newGuidJsonObj}]"
+                )
+            }
+        } catch (t: Throwable) {
+            config.logger.verbose(config.accountId, "Error migrating cached GUIDs: $t")
+            migrationSuccessful = false
+        }
+        return migrationSuccessful
+    }
+
+    /**
+     * This method migrates the encryption level of the user profiles stored in the local db
+     * Only pii data such as name, phone, email and identity are encrypted from the user profile, remaining are kept as is
+     *
+     * @param encrypt - Flag to indicate the task to be either encryption or decryption
+     * Returns true if migration was successful and false otherwise
+     */
+    private fun migrateDBProfile(
+        encrypt: Boolean
+    ): Boolean {
+        config.logger.verbose(
+            config.accountId,
+            "Migrating encryption level for user profiles in DB"
+        )
+
+        val profiles = dbAdapter.fetchUserProfilesByAccountId(config.accountId)
+
+        var migrationSuccessful = true
+        for ((deviceID, profile) in profiles) {
+            try {
+                piiDBKeys.forEach { piiKey ->
+                    profile.optString(piiKey).let { value ->
+                        val migrationResult =
+                            performMigrationStep(encrypt, value)
+                        migrationSuccessful =
+                            migrationSuccessful && migrationResult.migrationSuccessful
+                        profile.put(piiKey, migrationResult.data)
+                    }
+                }
+                if (dbAdapter.storeUserProfile(config.accountId, deviceID, profile) <= -1L) {
+                    migrationSuccessful = false
+                }
+            } catch (e: Exception) {
+                config.logger.verbose(config.accountId, "Error migrating profile $deviceID: $e")
+                migrationSuccessful = false
+            }
+        }
+
+        return migrationSuccessful
+    }
+
+
+    /**
+     * This method migrates the encryption level of the inapp data stored in the shared preferences file.
+     * Migration(if needed) is always performed to AES_GCM
+     *
+     * Returns true if migration was successful and false otherwise
+     */
+    private fun migrateInAppData(): Boolean {
+        config.logger.verbose(config.accountId, "Migrating encryption for InAppData")
+        var migrationSuccessful = true
+
+        val sharedPrefsDir = File(context.applicationInfo.dataDir, "shared_prefs")
+
+        // Fetch all SharedPreferences files starting with "inApp" and ending with the accountId
+        val prefsFiles = sharedPrefsDir.listFiles { _, name ->
+            name.startsWith("inApp") && name.endsWith("${config.accountId}.xml")
```

Comment:
Use Constants.INAPP_KEY instead of "inApp". Mention StoreProvider::constructStorePreferenceName to know how the name is constructed.

Reply:
Addressed in [bab4cbb](https://github.com/CleverTap/clevertap-android-sdk/pull/719/commits/bab4cbbdf777894e45e9521a0c53b5e005d7c6c0)